### PR TITLE
soften links

### PR DESCRIPTION
**Pull request by russelltsherman**

### CODE REVIEW

**Comment by russelltsherman** [comment]
now with less ENVVAR

File: app/Tickets/Flipside/class.FlipsideTicketRequestEmail.php [hunk 1]
```diff
@@ -41,7 +41,7 @@ public function getHTMLBody()
                 Reminder:  The dates for mailing your ticket request envelope are '.$this->settings['mail_start_date'].' through '
                 .$this->settings['request_stop_date'].'. Your postmark must be within that date range.<br><br>
                 If you need to update your ticket registration before mailing in your ticket request, return to 
-                <a href="https://secure.burningflipside.com/tickets/">secure.burningflipside.com/tickets/</a>.<br><br>
+                <a href="'.$this->secureUrl.'/tickets/">secure.burningflipside.com/tickets/</a>.<br><br>
```
Comment (pboyd04) [inline app/Tickets/Flipside/class.FlipsideTicketRequestEmail.php]: We should just make this consistent with the other emails from the ticket system (i.e. text is in the DB and admin editable).

File: app/Tickets/class.TicketPDF.php [hunk 2]
```diff
@@ -42,7 +42,7 @@ private function createPDFBody()
         //}
         $barcode_hash   = $tmp;
         $barcode        = '<barcode code="'.$barcode_hash.'" type="C128B"/>';
-        $transfer_qr    = '<barcode code="https://secure.burningflipside.com/tickets/transfer.php?id='.$this->ticket->hash.'" type="QR" class="barcode" size="1" error="M" />';
+        $transfer_qr    = '<barcode code="'.$this->secureURL..'/tickets/transfer.php?id='.$this->ticket->hash.'" type="QR" class="barcode" size="1" error="M" />';
```
Comment (pboyd04) [inline app/Tickets/class.TicketPDF.php]: Typo ".."

File: class.TicketPage.php [hunk 3]
```diff
@@ -67,7 +67,7 @@ function print_page($header=true)
         {
             $this->body = '
 <div id="content">
-    <h1>You must <a href="https://profiles.burningflipside.com/login.php?return='.$this->currentURL().'">log in <span class="fa fa-sign-in"></span></a> to access the Burning Flipside Ticket system!</h1>
+    <h1>You must <a href="'.getenv('PROFILES_URL').'/login.php?return='.$this->currentURL().'">log in <span class="fa fa-sign-in"></span></a> to access the Burning Flipside Ticket system!</h1>
```
Comment (pboyd04) [inline class.TicketPage.php]: Still an environment variable.

File: plugin.php [hunk 4]
```diff
@@ -12,12 +13,13 @@ function get_secure_menu_entries($page, $user)
         {
             $ret['Ticket System Data Entry']=$page->secure_root.'tickets/_admin/data.php';
         }
+        
         return $ret;
     }
 
     function get_plugin_entry_point()
     {
-        return array('name'=>'Ticket Registration/Transfer', 'link'=>'tickets/index.php');
```
Comment (pboyd04) [inline plugin.php]: We might want to have this switch text based on the part of the ticket window.

File: app/Tickets/Flipside/class.FlipsideTicketRequestEmail.php (hunk 1 above)
Comment (russelltsherman) [inline app/Tickets/Flipside/class.FlipsideTicketRequestEmail.php]: lets create a separate ticket to track that change.. 
it's not directly relevant to softening links

File: plugin.php (hunk 4 above)
Comment (russelltsherman) [inline plugin.php]: if we decide that's what we want then lets create a separate ticket to track that change.. 
it's not directly relevant to softening links

File: app/Tickets/Flipside/class.FlipsideTicketRequestEmail.php (hunk 1 above)
Comment (pboyd04) [inline app/Tickets/Flipside/class.FlipsideTicketRequestEmail.php]: I really don't like the Email and PDF base classes having any concept of the website or it's structure. I have part of a change to fix all that. Will push later this evening.

File: class.TicketPage.php [hunk 5]
```diff
@@ -67,7 +67,7 @@ function print_page($header=true)
         {
             $this->body = '
 <div id="content">
-    <h1>You must <a href="https://profiles.burningflipside.com/login.php?return='.$this->currentURL().'">log in <span class="fa fa-sign-in"></span></a> to access the Burning Flipside Ticket system!</h1>
+    <h1>You must <a href="'.$this->profilesUrl.'/login.php?return='.$this->currentURL().'">log in <span class="fa fa-sign-in"></span></a> to access the Burning Flipside Ticket system!</h1>
```
Comment (pboyd04) [inline class.TicketPage.php]: Should just use $this->loginUrl.'?return...

File: plugin.php (hunk 4 above)
Comment (pboyd04) [inline plugin.php]: I guess, my question is why the change at all?

File: _admin/class.TicketAdminPage.php [hunk 6]
```diff
@@ -111,7 +111,7 @@ function print_page($header = true)
             $this->body = '
         <div class="row">
             <div class="col-lg-12">
-                <h1 class="page-header">You must <a href="https://profiles.burningflipside.com/login.php?return='.$this->currentURL().'">log in <span class="fa fa-sign-in"></span></a> to access the Burning Flipside Ticket system!</h1>
+                <h1 class="page-header">You must <a href="'.$this->profilesUrl.'/login.php?return='.$this->currentURL().'">log in <span class="fa fa-sign-in"></span></a> to access the Burning Flipside Ticket system!</h1>
```
Comment (pboyd04) [inline _admin/class.TicketAdminPage.php]: We should just use the $this->loginUrl that already exists.

File: plugin.php (hunk 4 above)
Comment (russelltsherman) [inline plugin.php]: because i disliked the language flow of the menu
we don't register a ticket in the same way that we register an art car or a theme... 
and slashes in menu links are like contractions in titles..  best avoided..

File: app/Tickets/Flipside/class.FlipsideTicketRequestEmail.php (hunk 1 above)
Comment (russelltsherman) [inline app/Tickets/Flipside/class.FlipsideTicketRequestEmail.php]: what problem do you see with exposing our config object to the email and pdf classes.. which need to access the site config in order to expose the correct url links?

File: plugin.php (hunk 4 above)
Comment (russelltsherman) [inline plugin.php]: i'll revert this as it is not relevant to softening links..

File: app/Tickets/Flipside/class.FlipsideTicketRequestEmail.php (hunk 1 above)
Comment (pboyd04) [inline app/Tickets/Flipside/class.FlipsideTicketRequestEmail.php]: It just feels like excessive coupling of our framework to our current implementation. I don't like that. See https://github.com/BurningFlipside/Tickets/pull/19 and let me know if you think this solves your issue.

Comment (russelltsherman) [inline app/Tickets/Flipside/class.FlipsideTicketRequestEmail.php]: i guess i'm not clear on what you're labeling as being excessive coupling.. 
if it's the mailer accessing our settings.. pull #19 still has that going on..
if it's mail copy being hard coded into a class.. i completely agree..

Comment (pboyd04) [inline app/Tickets/Flipside/class.FlipsideTicketRequestEmail.php]: My issue is about the framework knowing about every site. So currently your pull added: /** The sites URLs */
 +    protected $wwwUrl;
 +    protected $wikiUrl;
 +    protected $profilesUrl;
 +    protected $secureUrl;

To email. If we deprecate the wiki we need to go yank wiki out of there. If we add a site called say "fred" we need to go add fred there.

The difference between this an #19 is that #19 is only grabbing the data it needs about its own site. While your proposed change grabs every site on every other site. It's not needed and adds extra complexity when/if we change the site structure.

Comment (russelltsherman) [inline app/Tickets/Flipside/class.FlipsideTicketRequestEmail.php]: if or when we actually deprecate the wiki then we will deal with the requirements of deprecating the wiki.. untill that day is near we should avoid using it as a reason for anything..

if you don't like exposing the extra vars.. OK.. but the complaint against that really belongs on the pull request which actually makes that change.. and this is not that PR..

this and all it's related pull requests are seeking to decouple the hard coded urls from the system to make the dockerized dev platform function properly.. 

your pull #19 is scope creep.. plain and simple.. you are blocking my forward momentum to introduce other changes in the system which are not directly relevant to my directive..

holding up these pull requests should only be done if you have a fundamental issue with the specific changes or implementation of my pull requests.. 

if you have issues with the underlying systems or historical implementations.. those should be handled later..

Comment (pboyd04) [inline app/Tickets/Flipside/class.FlipsideTicketRequestEmail.php]: I do have fundamental issues with the way this is going. If we're going to edit stuff, let's do it right. 

And I did complain about this in the other pull as well. You just didn't respond to that comment while you did to this one.

Comment (russelltsherman) [inline app/Tickets/Flipside/class.FlipsideTicketRequestEmail.php]: i see no unaddressed comments on any other PR.. 
i would expect your comments on this topic to be here.. https://github.com/BurningFlipside/CommonCode/pull/29

i want to do this right.. but i also want to focus on one task at a time.. 
pausing to refactor an existing implementation every time i draw your attention so some code isn't up to par..  is not in my opinion the right way.. 

if it isn't currently broken i want to do it later..

my primary goal is to complete the dev sandbox project.. then i'll be happy to circle back and address outstanding tickets on these repos..

**Comment by pboyd04** [comment]
I don't know why... here it is... been there for days...
[image: Untitled.png]

**Comment by russelltsherman** [comment]
image not shown

**Comment by pboyd04** [comment]
ugh... let me dig around see why you aren't seeing it...

**Comment by pboyd04** [comment]
Apparently I had to submit the review for some reason... weird...

**Comment by pboyd04** [comment]
Sorry, I thought you could see that other feedback. Still getting used to the github review interface.

**Comment by russelltsherman** [comment]
i still don't see that other feedback..
so how about this.. i need every last hard coded link in this set of applications to be configurable.. 
when you decide how that should happen let me know and i'll start moving forward with my project..

File: app/Tickets/Flipside/class.FlipsideTicketRequestEmail.php (hunk 1 above)
Comment (pboyd04) [inline app/Tickets/Flipside/class.FlipsideTicketRequestEmail.php]: Very weird. Can you check out https://github.com/BurningFlipside/CommonCode/pull/35 ? That addresses all my comments about https://github.com/BurningFlipside/CommonCode/pull/29 and I think it takes care of your issue while still allowing my dev box to work as well.

**Comment by pboyd04** [comment]
I've put up https://github.com/BurningFlipside/CommonCode/pull/35 for the CommonCode. I think that will solve both our issues and get us past this impasse. I really wish I knew why you couldn't see all that other feedback...

**Comment by pboyd04** [comment]
Waiting on resolution of https://github.com/BurningFlipside/CommonCode/pull/29 before merge.
